src/libAtomVM/ets_hashtable.c

```diff
@@ -82,8 +82,45 @@ static void print_info(struct EtsHashTable *hash_table)
 }
 #endif
 
-EtsHashtableErrorCode ets_hashtable_insert(struct EtsHashTable *hash_table, term key, term entry, EtsHashtableOptions opts, Heap *heap, GlobalContext *global)
+struct HNode *ets_hashtable_new_node(term entry, int keypos, GlobalContext *global)
 {
+    Heap *heap = malloc(sizeof(Heap));
```

Comment:
Small trick and optimization:
Heap struct can be embedded inside HNode this way:
```c
struct HNode
{
    struct HNode *next;
    term key;
    term entry;
    Heap heap;
};
```

Also, this will allow to avoid malloc overhead, simplifies code and also reduces memory fragmentation